DualityPlugins/Tilemaps/Tilemaps/Components/TilemapRenderer.cs

```diff
@@ -201,6 +201,58 @@ public Point2 GetTileAtLocalPos(Vector2 localPos, TilePickMode pickMode)
 			return tileIndex;
 		}
 		/// <summary>
+		/// Gets the local position of the specified tile. The function does not check if the point is a valid tile position.
+		/// </summary>
+		/// <param name="tile">The position of the tile.</param>
+		/// <param name="alignment">Determines the point of the tile that is returned.</param>
+		/// <returns></returns>
+		public Vector2 GetLocalPosAtTile(Point2 tile, Alignment alignment = Alignment.Center)
+		{
+			try
```

Comment:
Do not use try-catch to avoid `NullReferenceExceptions` - it's really, really inefficient compared to the alternative of just checking for null and avoiding access.

You should be able to get away with two null checks: `this.ActiveTilemap` and the `tileset` you get by accessing `this.ActiveTilemap.Tileset.Res`. This will also allow you to fall back to the default tile size. Suggestion: Mirror [this](https://github.com/AdamsLair/duality/blob/master/DualityPlugins/Tilemaps/Tilemaps/Components/TilemapRenderer.cs#L232-L236).